src/pages/dar_application/ProgressReportApplication.tsx

```diff
@@ -10,15 +11,17 @@ import SubmitProgressReport from 'src/pages/progress_reports/SubmitProgressRepor
 import {DataUseAcknowledgements} from 'src/pages/dar_application/DataUseAcknowlegements';
 import {translateDataUseRestrictionsFromDataUseArray} from 'src/libs/dataUseTranslation';
 import {validatePRFormData} from 'src/utils/darFormUtils';
-import {FormValidationState} from 'src/pages/dar_application/FormValidationState';
+import {FormValidationState, ValidationError} from 'src/pages/dar_application/FormValidationState';
 
 type ProgressReportApplicationProps = {
-    dar: DataAccessRequest, // corresponds either to the parent DAR for a new application or an existing readonly progress report
-    datasets: Dataset[],
-    readOnlyMode: boolean
+  readonly dar: DataAccessRequest, // corresponds either to the parent DAR for a new application or an existing readonly progress report
+  readonly datasets: Dataset[],
+  readonly readOnlyMode?: boolean
+  readonly location?: Location
```

Comment:
Why is location optional?

Reply:
It's not necessary in the read-only mode. Location is what we use to get back to the current page after redirecting to the NIH auth page.

Reply:
It sounds like we will have a number of objects that will only exist if `readOnlyMode` is false. Can we make this more clear? This can be represented in Typescript using discriminated union types, for example:

```
type ConditionalObject = 
  | { readOnly: false; propertyWhenReadOnly: string }
  | { readOnly: true; propertyWhenReadOnly?: never };
```

Another possibility is to have a separate object just for `nonReadOnly` types, that way it makes it easier to scan what is expected vs non-expected types.

```
missingWhenReadOnly: {
    location?: Location,
}
```